packages/client/hmi-client/src/components/Modal.vue

```diff
@@ -0,0 +1,79 @@
+<script setup lang="ts">
```

Comment:
Add a JSDoc style comment to explain the purpose of this component and how to use the slots, etc.